test/e2e/downwardapi_volume.go

```diff
@@ -92,6 +100,168 @@ var _ = Describe("Downward API volume", func() {
 			fmt.Sprintf("%s\n", podName),
 		}, f.Namespace.Name)
 	})
+
+	It("should verify update on labels modification", func() {
+		labels := map[string]string{}
+		labels["key1"] = "value1"
+		labels["key2"] = "value2"
+		podName := "labelsupdate" + string(util.NewUUID())
+		pod := &api.Pod{
+			ObjectMeta: api.ObjectMeta{
+				Name:   podName,
+				Labels: labels,
+			},
+			Spec: api.PodSpec{
+				Containers: []api.Container{
+					{
+						Name:    "client-container",
+						Image:   "gcr.io/google_containers/busybox",
+						Command: []string{"sh", "-c", "while true; do if [[ -e /etc/labels ]]; then cat  /etc/labels; fi; sleep 1; done"},
+						VolumeMounts: []api.VolumeMount{
+							{
+								Name:      "podinfo",
+								MountPath: "/etc",
+								ReadOnly:  false,
+							},
+						},
+					},
+				},
+				Volumes: []api.Volume{
+					{
+						Name: "podinfo",
+						VolumeSource: api.VolumeSource{
+							DownwardAPI: &api.DownwardAPIVolumeSource{
+								Items: []api.DownwardAPIVolumeFile{
+									{
+										Path: "labels",
+										FieldRef: api.ObjectFieldSelector{
+											APIVersion: "v1",
+											FieldPath:  "metadata.labels",
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+				RestartPolicy: api.RestartPolicyNever,
+			},
+		}
+
+		containerName := "client-container"
+		defer func() {
+			By("Deleting the pod")
+			f.Client.Pods(f.Namespace.Name).Delete(pod.Name, api.NewDeleteOptions(0))
+		}()
+		By("Creating the pod")
+		if _, err := f.Client.Pods(f.Namespace.Name).Create(pod); err != nil {
+			Failf("Failed to create pod: %v", err)
+		}
+		expectNoError(waitForPodRunningInNamespace(f.Client, pod.Name, f.Namespace.Name))
+
+		By("Sleeping to give kublet time to update volume")
```

Comment:
nit: kubelet

also, is there a better way of checking this than just a timeout? this seems prone to flakiness.